File: src/main/java/org/kite9/diagram/visualization/pipeline/ArrangementPipeline.java
package org.kite9.diagram.visualization.pipeline;

import org.kite9.diagram.model.Diagram;
import org.kite9.diagram.model.position.RenderingInformation;

public interface ArrangementPipeline {

	/**
	 * Performs the process of arranging elements on a diagram, giving them 
	 * all {@link RenderingInformation} elements so that they can be rendered.
	 */
	public abstract Diagram arrange(Diagram d);
}

File: kite9-visualization/src/commonMain/kotlin/org/kite9/diagram/visualization/pipeline/ArrangementPipeline.kt
package org.kite9.diagram.visualization.pipeline

import org.kite9.diagram.model.Diagram

interface ArrangementPipeline {
    /**
     * Performs the process of arranging elements on a diagram, giving them
     * all [RenderingInformation] elements so that they can be rendered.
     */
    fun arrange(d: Diagram): Diagram

}